client/pages/NewEvent.js

```diff
@@ -278,27 +280,28 @@ class NewEvent extends React.Component {
 
     const fromMinutes = getMinutes(fromTime);
     const toMinutes = getMinutes(toTime);
-
+    // create a date range as date
     if (dateOrDay) {
       const dates = [];
 
-      for (const key of Object.keys(weekDays)) {
-        if (!weekDays[key]) continue;
-        dates.push({
-          fromDate: moment()
+      Object.keys(weekDays).forEach((key) => {
```

Comment:
I prefer the previous iteration of the code because there is one less level of indentation in the code. for..of loops are disabled in eslint because "some users" don't like them. We can add a rule in the .eslintrc to stop eslint from red-flagging them.